syntax = "proto3";

package proto;
option go_package = "github.com/hashicorp/go-plugin/examples/kv/proto";

service KV {
  rpc Get(GetRequest) returns (GetResponse);
  rpc Put(PutRequest) returns (Empty);
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  bytes value = 1;
}

message PutRequest {
  string key = 1;
  bytes value = 2;
}

message Empty {}
